cpp/src/arrow/util/bit-util.h

```diff
@@ -166,148 +115,93 @@ static inline bool IsMultipleOf64(int64_t n) { return (n & 63) == 0; }
-  return (value / factor) * factor;
-}
-
-/// Returns 'value' rounded up to the nearest multiple of 'factor' when factor is
-/// a power of two
-static inline int RoundUpToPowerOf2(int value, int factor) {
+// Returns 'value' rounded up to the nearest multiple of 'factor' when factor
+// is a power of two.
+// The result is undefined on overflow, i.e. if `value > 2**64 - factor`,
+// since we cannot return the correct result which would be 2**64.
+static inline int64_t RoundUpToPowerOf2(int64_t value, int64_t factor) {
   // DCHECK((factor > 0) && ((factor & (factor - 1)) == 0));
   return (value + (factor - 1)) & ~(factor - 1);
 }
 
-static inline int RoundDownToPowerOf2(int value, int factor) {
-  // DCHECK((factor > 0) && ((factor & (factor - 1)) == 0));
-  return value & ~(factor - 1);
-}
-
-/// Specialized round up and down functions for frequently used factors,
-/// like 8 (bits->bytes), 32 (bits->i32), and 64 (bits->i64).
-/// Returns the rounded up number of bytes that fit the number of bits.
-static inline uint32_t RoundUpNumBytes(uint32_t bits) { return (bits + 7) >> 3; }
-
-/// Returns the rounded down number of bytes that fit the number of bits.
-static inline uint32_t RoundDownNumBytes(uint32_t bits) { return bits >> 3; }
-
-/// Returns the rounded up to 32 multiple. Used for conversions of bits to i32.
-static inline uint32_t RoundUpNumi32(uint32_t bits) { return (bits + 31) >> 5; }
-
-/// Returns the rounded up 32 multiple.
-static inline uint32_t RoundDownNumi32(uint32_t bits) { return bits >> 5; }
-
-/// Returns the rounded up to 64 multiple. Used for conversions of bits to i64.
-static inline uint32_t RoundUpNumi64(uint32_t bits) { return (bits + 63) >> 6; }
-
-/// Returns the rounded down to 64 multiple.
-static inline uint32_t RoundDownNumi64(uint32_t bits) { return bits >> 6; }
-
-template <int64_t ROUND_TO>
-static inline int64_t RoundToPowerOfTwo(int64_t num) {
-  // TODO(wesm): is this definitely needed?
-  // DCHECK_GE(num, 0);
-  constexpr int64_t force_carry_addend = ROUND_TO - 1;
-  constexpr int64_t truncate_bitmask = ~(ROUND_TO - 1);
-  constexpr int64_t max_roundable_num = std::numeric_limits<int64_t>::max() - ROUND_TO;
-  if (num <= max_roundable_num) {
-    return (num + force_carry_addend) & truncate_bitmask;
-  }
-  // handle overflow case.  This should result in a malloc error upstream
-  return num;
-}
-
-static inline int64_t RoundUpToMultipleOf64(int64_t num) {
-  return RoundToPowerOfTwo<64>(num);
-}
-
 static inline int64_t RoundUpToMultipleOf8(int64_t num) {
-  return RoundToPowerOfTwo<8>(num);
-}
-
-/// Non hw accelerated pop count.
-/// TODO: we don't use this in any perf sensitive code paths currently.  There
-/// might be a much faster way to implement this.
-static inline int PopcountNoHw(uint64_t x) {
-  int count = 0;
-  for (; x != 0; ++count) x &= x - 1;
-  return count;
-}
-
-/// Returns the number of set bits in x
-static inline int Popcount(uint64_t x) {
```

Comment:
We will want to popcount at some point (e.g. computing null counts fast on validity bitmaps). But I guess we can add new implementations when the time comes

Reply:
popcount is already used for computing null counts in `CountSetBits`. But those specific wrappers were unused.